pkg/jobqueue/dbjobqueue/dbjobqueue.go

```diff
@@ -904,9 +913,148 @@ func (q *DBJobQueue) jobDependents(ctx context.Context, conn connection, id uuid
 	return dependents, nil
 }
 
+// deleteJob removes the job from the database
+// the CASCADE constraint will also delete any entries from the job_dependencies table
+func (q *DBJobQueue) deleteJob(jobID uuid.UUID) error {
+	conn, err := q.pool.Acquire(context.Background())
```

Comment:
This API is a mixed bag when it comes to context passing, some methods do support it, some don't. Thing is, I am working on migrating from `logrus` to `log/slog` and I will be refactoring everything to make use of context passing. Would you mind adding `ctx context.Context` function argument for all newly introduced methods already here so I have less work later? Just pass the context down the stack to all helpers, the benefit will be that all the logging coming from the SQL driver will have correlation id so we can search more easily for detailed logs.